xla/service/gpu/runtime3/sync_on_streams_thunk.h

```diff
@@ -0,0 +1,57 @@
+/* Copyright 2024 The OpenXLA Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef XLA_SERVICE_GPU_RUNTIME3_SYNC_ON_STREAMS_THUNK_H_
+#define XLA_SERVICE_GPU_RUNTIME3_SYNC_ON_STREAMS_THUNK_H_
+
+#include <string>
+
+#include "absl/status/status.h"
+#include "xla/service/gpu/thunk.h"
+
+namespace xla {
+namespace gpu {
+
+// This thunk
+class SyncOnStreamsThunk : public Thunk {
```

Comment:
`source_stream->ThenWaitFor(stream)` impl makes `WaitForStreamsThunk` a much better name

Reply:
yea.. i'd agree with that :). updated it